definition/definition.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2017 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package definition
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/caicloud/nirvana/router"
+)
+
+type Operator func(ctx context.Context, object interface{}) (interface{}, error)
+
+type Method string
+
+const (
+	List        Method = "List"
+	Get         Method = "Get"
+	Create      Method = "Create"
+	Update      Method = "Update"
+	Patch       Method = "Patch"
+	Delete      Method = "Delete"
+	AsyncCreate Method = "AsyncCreate"
+	AsyncUpdate Method = "AsyncUpdate"
+	AsyncPatch  Method = "AsyncPatch"
+	AsyncDelete Method = "AsyncDelete"
+)
+
+type Source string
+
+const (
+	Path   Source = "Path"
+	Query  Source = "Query"
+	Header Source = "Header"
+	Form   Source = "Form"
+	File   Source = "File"
+	Body   Source = "Body"
+	Prefab Source = "Prefab"
+	Auto   Source = "Auto"
+)
+
+type Kind string
+
+const (
+	Integer   Kind = "Integer"
+	Integers  Kind = "Integers"
+	Number    Kind = "Number"
+	Numbers   Kind = "Numbers"
+	String    Kind = "String"
+	Strings   Kind = "Strings"
+	Binary    Kind = "Binary"
+	Boolean   Kind = "Boolean"
+	Booleans  Kind = "Booleans"
+	Date      Kind = "Date"
+	Dates     Kind = "Dates"
+	DateTime  Kind = "DateTime"
+	DateTimes Kind = "DateTimes"
+	Object    Kind = "Object"
+	Objects   Kind = "Objects"
+)
+
+type Type string
+
+const (
+	Meta  Type = "Meta"
+	Data  Type = "Data"
+	Error Type = "Error"
+)
+
+type Example struct {
+	Description string
+	Instance    interface{}
+}
+
+type Parameter struct {
+	Source Source
+	Name   string
+	// Optional. If you want to override the type in function parameter,
+	// you can specify it here.
+	Type      reflect.Type
```

Comment:
I think this `Type` will add extra burden for users...
Since it was used to handle API compatibility which is uncommon in normal cases.